src/VecSim/vec_sim_common.h

```diff
@@ -193,16 +193,26 @@ typedef struct {
                       // to get it from the parameters resolve function.
 } VecSimQueryParams;
 
+/**
+ * Index info that is static and immutable (cannot be changed over time)
+ */
 typedef struct {
-    size_t indexSize;        // Current count of vectors.
-    size_t indexLabelCount;  // Current unique count of labels.
-    size_t blockSize;        // Brute force algorithm vector block (mini matrix) size
-    VecSimMetric metric;     // Index distance metric
-    uint64_t memory;         // Index memory consumption.
-    VecSimType type;         // Datatype the index holds.
-    bool isMulti;            // Determines if the index should multi-index or not.
-    size_t dim;              // Vector size (dimension).
-    VecSearchMode last_mode; // The mode in which the last query ran.
+    VecSimAlgo algo;     // Algorithm being used.
+    size_t blockSize;    // Brute force algorithm vector block (mini matrix) size
+    VecSimMetric metric; // Index distance metric
+    VecSimType type;     // Datatype the index holds.
+    bool isMulti;        // Determines if the index should multi-index or not.
+    size_t dim;          // Vector size (dimension).
+
+    VecSimAlgo tieredBackendAlgo; // The algorithm for the tiered index (if algo is tiered).
```

Comment:
I think it would be clearer and more meaningful to have `algo` as the primary algorithm, and to have `isTiered` as a bool

Reply:
why the backend algorithm is relevant here? we have it at `tieredInfoStruct`

Reply:
I needed it in the concise `VecSimIndexStaticInfo` struct because we use it in `VecSimIndex_ResolveParams` (which is called in every query).

Reply:
@GuyAv46  I like this idea